persist-tool/add.c

```diff
@@ -210,8 +207,9 @@ add_main(int argc, char *argv[])
       fprintf(stderr, "Directory doesn't exist: %s\n", persist_state_dir);
       return 1;
     }
-  filename = g_build_path(G_DIR_SEPARATOR_S, persist_state_dir,
-                          persist_state_name ? persist_state_name : DEFAULT_PERSIST_FILE, NULL);
+
+  gchar *filename = g_build_path(G_DIR_SEPARATOR_S, persist_state_dir,
+                                 persist_state_name ? persist_state_name : DEFAULT_PERSIST_FILE, NULL);
```

Comment:
Maybe a goto-based resource handling would be better...

Reply:
Right, fixed.

Reply:
:)